volume-providers/vpc/provider/util.go

```diff
@@ -86,6 +90,34 @@ func skipRetry(err *models.Error) bool {
 	return false
 }
 
+// skipRetryForIKS skip retry as per listed error codes
+func skipRetryForIKS(err *models.IksError) bool {
+	skipStatus, ok := skipErrorCodes[string(err.Code)]
+	if ok {
+		return skipStatus
+	}
+	return false
+}
+
+// skipRetryForAttach skip retry as per listed error codes
+func skipRetryForAttach(err error, isIKS bool) bool {
+	// Only for storage-api ms related calls error
+	if isIKS {
+		iksError, ok := err.(*models.IksError)
+		if ok {
+			return skipRetryForIKS(iksError)
```

Comment:
we can call `skipRetry` here and that should work as well ideally.

Reply:
iksError is of type `models.IksError` and other method have different type